Automattic/wp-calypso
# WooCommerce: Enable refunds on orders

This PR splits the OrderDetails component into `OrderDetailsTable` and `OrderRefundCard`. In `OrderRefundCard`, I've added a refund modal so that store owners can refund customers.

Screenshot of the modal, where a user can refund based on items quantities, shipping cost, or enter a custom refund amount:

![screen shot 2017-06-27 at 7 43 04 pm](https://user-images.githubusercontent.com/541093/27615200-9a55fa98-5b74-11e7-9ab6-892a871771d2.png)

If they try to refund more than the cost of the order, they see an error:

![screen shot 2017-06-27 at 8 07 17 pm](https://user-images.githubusercontent.com/541093/27615199-9a506790-5b74-11e7-8deb-4a3f16ec7803.png)

If the refund is successful, there's a global notice after the modal closes (if the refund fails on the server side, it's an error notice), and the order details update with the new refund amount.

![screen shot 2017-06-27 at 8 18 54 pm](https://user-images.githubusercontent.com/541093/27615404-f28d5e1c-5b75-11e7-90ca-9095baffb99f.png)

Left to do:

- ~Entering in a custom refund amount is broken right now, either I'll fix this or convert it to read only.~ Custom amount has been switched to read-only, formatting the price breaks the input
- ~The credit card display needs to be hooked up.~ See https://github.com/Automattic/wp-calypso/pull/15589#issuecomment-311780309, this will be coming out and replaced with payment method info
- ~A complete refund should toggle the order status to refunded.~ Does this automatically 🎉

**To test**

1. Make sure you have [the latest master API plugin](https://github.com/woocommerce/wc-api-dev) on your site
2. Visit your orders list `/store/orders/:site`
3. Click into an order
4. Click the "Submit Refund" button
5. Create a refund, try with various refund amounts (0, higher than total, partial refund, exact...)

Run `npm test` to test the new action/reducer/selector (for tracking refund requests).

Fixes #15672
